### ISSUE TITLE
Export properties as IfcPropertyEnumeratedValue

### ISSUE THREAD

**Issue by eibre** [Autodesk/revit-ifc#250]
Is it possible to export a paramter as an `IfcPropertyEnumeratedValue`?

For instance from https://standards.buildingsmart.org/IFC/RELEASE/IFC4/ADD2_TC1/HTML/schema/ifcproductextension/pset/penum_elementstatus.htm

See this post on Building Smart forum: https://forums.buildingsmart.org/t/phasing-in-ifc/3155/3

**Comment by AngelVelezSosa** [comment]
That should work. What version of Revit are you on?

**Comment by eibre** [comment]
I tested this with 2020:

![image](https://user-images.githubusercontent.com/5968613/100618674-716cf900-331c-11eb-918d-142b531bcc87.png)

**Comment by WawanSolihin** [comment]
Can you send me the sample project? I see in the latest (not released yet), it exports the enumeration

**Comment by eibre** [comment]
@WawanSolihin  attached here. 
[Project1.zip](https://github.com/Autodesk/revit-ifc/files/5616478/Project1.zip)

**Comment by WawanSolihin** [comment]
I believe it is because you have parameter mapping from "Phase Created" to Status in Pset_WallCommon. The internal processing for Pset_WallCommon will consider Status using Enumeration, but parameter mapping will override it. Since "Phase Created" is an internal/built-in parameter, only the same will be taken and assigned to Status. The values in those parameters are NOT the same as the enumeration. I will suggest to remove the mapping and create "Status" parameter and use the appropriate/valid enumeration string values for PEnum_ElementStatus (see attached modified model and the export result)
[Project1-modified.zip](https://github.com/Autodesk/revit-ifc/files/5624046/Project1-modified.zip)

**Comment by tsolsen** [comment]
Why is the parameter not the same as the enum?

I would advice against proposing the suggestion, as it's reliant on a manual operation or plugin run - and thus a quality risk. We are already over reliant on dynamo runs, to perform lookups and formulas between parameters, as is.

The preferred solution should be a UI that allow us to nominate the built-in parameters as attributes and enums.

**Comment by AngelVelezSosa** [comment]
Phase created in Revit is a similar - but not identical - concept as Phase in IFC.  I think we should try to by default map Phase created to phase in IFC (there is only one phase in Pset_*Common - is created the right one?)  but it isn't a perfect fit.

**Comment by tsolsen** [comment]
I'd say created is the best fit to status.

**Comment by WawanSolihin** [comment]
Well the values from "Phase Created" do not conform to the PEnum_ElementStatus. So even it is mapped correctly, it cannot be exported correctly since the enum values are different

**Comment by eibre** [comment]
Thanks for the solution. It would be good to be able to map the values to Revits phasing system, I think something like this could work: 
![image](https://user-images.githubusercontent.com/5968613/100777461-23ccbb00-3406-11eb-8e8f-88b60a2d0a29.png)

**Comment by eibre** [comment]
Temporary would be `Phase Created == Phase to Export && Phase Demolished == Phase to Export` right?

**Comment by AngelVelezSosa** [comment]
Yes, I think it would work.  @WawanSolihin : we could have a StatusCalculator that would take the phase created, phase demolished of the element and phase to export and map to IFC.

**Comment by jokopperud** [comment]
Would really like to see the export of Phase solved. Todays workaround is manual and tediouis!!

**Comment by WawanSolihin** [comment]
I will create backlog item for this.

Wawan
